taojieTan/assignments/slidingwindow/lc904/lc904.py

```diff
@@ -7,13 +7,54 @@
 # Write a function to return the maximum number of fruits in both baskets.
 
 def fruits_into_baskets(fruits):
+  #input: an array with only characters
+  #output: return an integer 
 
+  #edge cases: 1. empty array -> 0 
 
-def main():
-  print("Maximum number of fruits: " + str(fruits_into_baskets(['A', 'B', 'C', 'A', 'C'])))
-  print("Maximum number of fruits: " + str(fruits_into_baskets(['A', 'B', 'C', 'B', 'B', 'C'])))
+  #initialize a dictionary to represent the basket and keep track of the frequency of distinct characters
+  basket = {}
+
+  #initialize two window pointers to keep track of the size of the sliding window
+  window_start = 0
+  window_end = 0
+
+  #initialize a max_fruits variable to keep track of the maximum number of fruits in the basket
+  max_fruits = 0
+
+  #loop through the array to store each character into basket
+  #if the length of basket is greater than 2 which means we just picked a third type of fruit, 
+  #we need to subtract the frequency of leftmost value (with window_start index) by one, 
+  #and in the meanwhile shifting the window_start index to right
+  #we keep subtracting until the length of basket is smaller or equal to two (use a while loop)
+  #after break out of the inner while loop, we compare larger length 
+  #after going through the entire array, we will get the final result of maximum length
```

Comment:
Love the psuedo-coding. This may be a personal preference, but psuedo-code at the top tends to look cleaner. Comments in-line are more for clarifying statements about the code rather than the explanation of the code itself.